screens/ExpenseTrackerScreen.js

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { AsyncStorage, Button, FlatList, StyleSheet, Text, TextInput, View } from 'react-native';
+
+import Colors from '../constants/Colors';
+
+export default function ExpenseTrackerScreen() {
+  const [expenses, setExpenses] = useState([]);
+  const [inputValues, setInputValues] = useState({ title: 'Expense', amount: '100' });
+  const flatList = useRef();
+
+  // read all expenses from local storage
+  const rehydrate = async () => {
+    const storedExpenses = JSON.parse(await AsyncStorage.getItem('expenses'));
+    setExpenses(storedExpenses ? storedExpenses : []);
+  };
+
+  const addExpense = () => {
+    if (inputValues.title && inputValues.amount) {
+      // determine the latest expense to get the id from which we need to increment for the new one
+      const latestExpense = expenses[expenses.length - 1];
+      const newExpense = {
+        id: latestExpense ? latestExpense.id + 1 : 1, // if there is no expense yet, start with 1
+        title: inputValues.title,
+        amount: inputValues.amount
+      };
+
+      // update expenses with the new expense
+      const updatedExpenses = [...expenses, newExpense];
+
+      setExpenses(updatedExpenses);
+    }
+  };
+
+  // run once (component did mount)
+  useEffect(() => {
+    rehydrate();
+  }, []);
+
+  // run always when expenses changed (component did update)
+  useEffect(() => {
+    // write all expenses to local storage
+    if (expenses && expenses.length) {
+      AsyncStorage.setItem('expenses', JSON.stringify(expenses));
+    }
+
+    // scroll the list to top after adding an expense
+    setTimeout(() => {
+      flatList && flatList.current.scrollToEnd();
+    }, 200);
+  }, [expenses]);
```

Comment:
The duration is 200ms, which is 0,2 seconds. 2 seconds would be 2000ms.
The time is chosen randomly. There is a timeout necessary, because otherwise it could scroll before the new expense is added. Using a timeout is not the best solution here, because we cannot be sure, that a new expense is added always within 200ms. It would be better to wait for a new expense to be added. I will add a TODO: in the code.
Need to be sure that `AsyncStorage.setItem('expenses', JSON.stringify(expenses));` has finished, before scrolling.

Reply:
TODO: is added with this commit: https://github.com/donni106/react-native-playground-app/pull/1/commits/0037d4d839281bb896dae1841f76dd79a8145d03

Reply:
Update with async-await instead of using timeout is added with this commit: https://github.com/donni106/react-native-playground-app/pull/1/commits/6a8998b92211b4a0f1a1cafc0c038e29217fe7b2